server/src/main/java/org/elasticsearch/index/mapper/DateFieldMapper.java

```diff
@@ -581,6 +598,13 @@ protected DateFieldMapper clone() {
 
     @Override
     protected void parseCreateField(ParseContext context) throws IOException {
+        if (singletonDataStreamTimestamp) {
+            if (context.isDataStreamTimestampParsed()) {
+                throw new IllegalArgumentException("data stream timestamp field [" + name() + "] encountered multiple values");
+            }
+            context.setDataStreamTimestampParsed(true);
```

Comment:
I just noticed that we should probably only set this after the date has been successfully parsed? Just writing this down to not forget, I see that we may be changing the strategy and moving to a metadata field mapper.